contrail-controller/hooks/contrail_controller_utils.py

```diff
@@ -179,3 +185,66 @@ def update_charm_status():
         docker_utils.compose_run(REDIS_CONFIGS_PATH + "/docker-compose.yaml")
 
     common_utils.update_services_status(SERVICES)
+
+
+def update_hosts_file(hosts_map):
+    """Update /etc/hosts file with cluster names and IPs.
+
+    RabbitMQ requires NODE names in a cluster to be resolvable.
+    https://www.rabbitmq.com/clustering.html#issues-hostname
+
+    In a multi-homed host scenario cluster IPs may have FQDNs structured
+    as interface_name.host_name.domain which will result in an issue if
+    a short name is derived from an FQDN by taking its first part.
+    See https://github.com/Juniper/contrail-charms/issues/50
+
+    This function updates /etc/hosts file with resolutions for IP -> hostname
+    lookups.
+    """
+    with open(HOSTS_FILE, 'r') as hosts:
+        lines = hosts.readlines()
+
+    log("Updating hosts file with: %s (current: %s)" % (hosts_map, lines),
+        level=INFO)
+
+    newlines = []
+    for ip, hostname in hosts_map.items():
+        if not ip or not hostname:
+            continue
+
+        keepers = []
+        for line in lines:
+            _line = line.split()
+            if len(line) < 2 or not (_line[0] == ip or hostname in _line[1:]):
+                keepers.append(line)
+            else:
+                log("Removing line '%s' from hosts file" % (line))
+
+        lines = keepers
+        newlines.append("%s %s\n" % (ip, hostname))
```

Comment:
is it sufficient to store only one form in /etc/hosts ? should it add short and fqdn forms?